apiv2/internal/server/host.go

```diff
@@ -719,54 +706,84 @@ func (is *InventorygRPCServer) GetHostsSummary(
 	var errorState uint32
 	var runningState uint32
 	var unallocatedState uint32
+	reqFilter := req.GetFilter()
+
+	filterIsFailedHostStatus := `%s = %s OR %s = %s OR %s = %s OR %s.%s = %s OR %s.%s = %s OR %s.%s = %s OR %s.%s = %s`
+	filterIsFailedHostStatus = fmt.Sprintf(filterIsFailedHostStatus,
+		inv_computev1.HostResourceFieldHostStatusIndicator,
+		api.HostResourceHostStatusIndicatorSTATUSINDICATIONERROR,
+		inv_computev1.HostResourceFieldOnboardingStatusIndicator,
+		api.HostResourceHostStatusIndicatorSTATUSINDICATIONERROR,
+		inv_computev1.HostResourceFieldRegistrationStatusIndicator,
+		api.HostResourceHostStatusIndicatorSTATUSINDICATIONERROR,
+		inv_computev1.HostResourceEdgeInstance,
+		inv_computev1.InstanceResourceFieldInstanceStatusIndicator,
+		api.HostResourceHostStatusIndicatorSTATUSINDICATIONERROR,
+		inv_computev1.HostResourceEdgeInstance,
+		inv_computev1.InstanceResourceFieldProvisioningStatusIndicator,
+		api.InstanceResourceInstanceStatusIndicatorSTATUSINDICATIONERROR,
+		inv_computev1.HostResourceEdgeInstance,
+		inv_computev1.InstanceResourceFieldUpdateStatusIndicator,
+		api.InstanceResourceInstanceStatusIndicatorSTATUSINDICATIONERROR,
+		inv_computev1.HostResourceEdgeInstance,
+		inv_computev1.InstanceResourceFieldTrustedAttestationStatusIndicator,
+		api.InstanceResourceInstanceStatusIndicatorSTATUSINDICATIONERROR,
+	)
+	filterInstanceRunning := `%s.%s = %q AND NOT (%s)`
+	filterInstanceRunning = fmt.Sprintf(filterInstanceRunning,
+		inv_computev1.HostResourceEdgeInstance,
+		inv_computev1.InstanceResourceFieldInstanceStatus,
+		"Running",
```

Comment:
I believe we are changing the behaviour, we were checking the CurrentState not the Status earlier.

Reply:
Fixed to check the state.